### ISSUE TITLE
Command-line DotNet commands not found under Windows 10

### ISSUE THREAD

**Issue by Petermarcu** [dotnet/core-setup#3281]
@marcusts commented on [Thu Aug 17 2017](https://github.com/dotnet/core/issues/854)

I followed all f the installation instructions, but am not able to run command such as:

$ dotnet new lib -o mylibrary

ETC., ETC.

Shall I add a path to my environment?  It would be ideal if you made this more clear.

---

@svick commented on [Thu Aug 17 2017](https://github.com/dotnet/core/issues/854#issuecomment-323040480)

If you successfully followed the installation instructions, then `dotnet` will be added to the PATH automatically, you shouldn't need to do anything else.

What is the error you're getting? What instructions did you follow? Did you restart your command line? Which command line are you using?

---

@marcusts commented on [Thu Aug 17 2017](https://github.com/dotnet/core/issues/854#issuecomment-323199179)

I am a client-server programmer working with Visual Studio 15.3 in Windows 1.  I did indeed follow all of the instructions . The name is not known when typing it from a command line, so there is no path to that executable.

---

@colinbowern commented on [Thu Aug 17 2017](https://github.com/dotnet/core/issues/854#issuecomment-323225642)

I am seeing this as well - Windows 10 x64. Looking at the install logs in temp there is no indication of path variable change. I do see the Environment parameter in the Wix source however:

https://github.com/dotnet/core-setup/blob/6887ab556bc8302390782711dcdd75b75e769cf5/src/pkg/packaging/windows/host/host.wxs

Any way to unpack the installer executable to get at the MSI or pass along verbose logging parameters?

---

@Petermarcu commented on [Wed Aug 30 2017](https://github.com/dotnet/core/issues/854#issuecomment-326072963)

@rakeshsinghranchi can you help with verbose logging instructions. After installation, `dotnet` should be on the path.

---

@rakeshsinghranchi commented on [Wed Aug 30 2017](https://github.com/dotnet/core/issues/854#issuecomment-326099693)

Here are the details : 

**.NET Core 2.0.0 x64 installer bundle:**
 http://dotnetcli.blob.core.windows.net/dotnet/Runtime/2.0.0/dotnet-runtime-2.0.0-win-x64.exe

MSI Verbose logging : http://wixtoolset.org/documentation/manual/v3/howtos/general/get_a_log.html

**To extract a bundle package :**
Extract wix toolset: https://github.com/wixtoolset/wix3/releases/download/wix311rtm/wix311exe.zip
Locate dark.exe and run below command :
   dark.exe -x <targetDir> <PathToBundlePackage>

**To collect VS setup log :** 
Download and run the collection tool - https://aka.ms/vscollect. Instruction is on the page.

---

@Petermarcu commented on [Fri Oct 06 2017](https://github.com/dotnet/core/issues/854#issuecomment-334828099)

Is anyone that is hitting this issue able to follow the logging instructions so we can try to figure out whats going on?

**Comment by boban984** [comment]
Reinstalled dotnet core SDK and now lost the CLI :) i guess path variable is missing

**Comment by Giovanni-Mattucci** [comment]
Yeah I had the same issue after using the **.NET Core 2.1.2 x64 installer**, where the path variable no longer worked, and caused an error in vscode (Related OmniSharp/omnisharp-vscode#627): 
`The .NET CLI tools cannot be located`.

For anyone running into this issue, I added back the path manually and the error has gone away ([stole these steps from java's page](https://www.java.com/en/download/help/path.xml)) :
**For Windows 10 and Windows 8:**
1. In Search, search for and then select: System (Control Panel)
2. Click the Advanced system settings link.
3. Click Environment Variables. In the section System Variables, find the PATH environment variable and select it. Click Edit. If the PATH environment variable does not exist, click New.
4. In the Edit System Variable (or New System Variable) window, specify the value of the PATH environment variable. Click OK. Close all remaining windows by clicking OK.
5. Reopen Command prompt window, and run `dotnet`.

**Comment by SaswatPadhi** [comment]
Same here. `dotnet` not found!

Installed 2.1.4 on Windows 10 64-bit.

**Comment by SaswatPadhi** [comment]
Sorry, works as expected on new PowerShell sessions.

I tried `refreshenv` on the ones that were running, and hoped they would pick up the new `PATH`, but they didn't.

**Comment by rakeshsinghranchi** [comment]
@SaswatPadhi , can you please share steps to reproduce the issue ? Are you installing SDK 2.1.4 using SDK installer link or via Visual Studio ?

**Comment by SaswatPadhi** [comment]
Hi,

I installed the 2.1.4 SDK from the package available online, not via VS.

> > Sorry, works as expected on new PowerShell sessions.
I tried refreshenv on the ones that were running, and hoped they would pick up the new PATH, but they didn't.

> Does opening a new command window resolve the issue ?

Yes, works fine on new PowerShell windows, but `refreshenv` didn't update the PATH.

Thanks.

**Comment by rakeshsinghranchi** [comment]
@SaswatPadhi ,  can you please open a bug for https://github.com/PowerShell/PowerShell on why _refreshenv_ didn't update the  PATH? This seems unrelated to .NET Core setup.

**Comment by jamesburton** [comment]
I had the same problem, with the latest v2.1 update, and it was resolved by adding "c:\Program Files\dotnet" as a PATH environment variable entry.

**Comment by Allann** [comment]
Same here, I had installed 2.2.100 preview3, came across a blocking issue and therefore uninstalled it and all other .Net Core SDK packages on the machine, therefore none existed.  Did a reboot, then installed dotnet-sdk-2.1.403-win-x64.exe  after which the Path variable no longer had dotnet in it.
And before you ask, I can't provide more information about repro-ing the issue than that.  I had most sdk version installed from 1.0 previews till 2.2.100 preview 3.  I thought that was the original issue hence why I uninstalled ALL of them and started afresh.
2.1.403 did install everything as expected except the path from all appearances because like others, when i manual add it all appears fine.  
Not sure if it might have something to do with the issue, but I have written my own command for dotnet and have the following in my path already `D:\Users\Allan\.dotnet\tools` so if the logic to determine if the path already has dotnet in it might be inspecting the value incorrectly.
